### PR TITLE
Asus Aura Implementation 02

### PR DESCRIPTION
**Pull request by Fluto**
**This pull request proposes the following changes:**
- Removes old Asus implementation as it does not quite work
- Introduce AuraSDK2 support, using Asus's guidelines on how to implement it
- Asus ROG Strix Flare Support (this time with extra LEDs and a new layout)

**Known issues/To do:**
- AuraSDK2 Is known to be buggy, so this needs a lot of testing before it is merged in
- 
-

### CODE REVIEW

**Comment by Fluto** [comment]
https://cdn.discordapp.com/attachments/542047451201404943/680956713905553453/2020-02-23_12-58-18.mp4

An example of how to set up custom keys for peripherals

**Review by xQwexx — CHANGES_REQUESTED** [review]

File: Project-Aurora/Project-Aurora/Devices/Asus/AsusHandler.cs [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+        public void Stop()
+        {
+            lock (deviceLock)
+            {
+                foreach (var device in devices)
+                    device.Stop();
+            
+                devices.Clear();
+                AuraSdk.ReleaseControl(0);
```
[the remark this answers is not in the document]
Comment (Fluto) [reply Project-Aurora/Project-Aurora/Devices/Asus/AsusHandler.cs]: good pickup

**Comment by xQwexx** [comment]
Could you add support for asus scope(sdk name: Charm), the sdk keycodes are not working for some the ctrl and windows key, and the asus sdk not supporting the uk layout just the us one,
I already done it in this pr: [https://github.com/diogotr7/Aurora/pull/4](https://github.com/diogotr7/Aurora/pull/4)

**Comment by Fluto** [comment]
I'll look into it, I have Strix Flare. The internal name for it is also charm :/
Perhaps you have a different LED count to differentiate them? What's yours? I believe if you check the log when running my build should tell you

**Comment by xQwexx** [comment]
Then it is look like there is no difference for the 2 keyboard from the sdk perspective. :( Than I think the only way to differentiate them is to put multiple layout in aurora and check which one is selected.
I sending the my device information:
image

**Comment by jdcmarques** [comment]
Hi, can't seem to get my Claymore to work using the build generated by AppVeyor...
Is there anyway I can help?

**Comment by Fluto** [comment]
@xQwexx I added the RIGHT_WINDOWS key (on my keyboard it's FN), but the ctrl codes seem to match so im now sure what's going on there

**Comment by Fluto** [comment]
@jdcmarques That might be because of your version of Aura that is installed. There is a guide to get that correctly setup on the Discord channel #asus-aura on the Aurora Discord Server.

**Comment by xQwexx** [comment]
So there is two thing first sdk hashcodes not support the uk layout so DeviceKeys.HASHTAG and DeviceKeys.BACKSLASH_UK not working so you need to set them manually, like this.
```
            //UK keyboard Layout
            if (keyCode == 0x56)
            {//Hashtag Key
                light = keyboard.Lights[(int)(3 * keyboard.Width + 13)];
            }
            else if (keyCode == 0x59)
            {//BackSlash Key
                light = keyboard.Lights[(int)(4 * keyboard.Width + 1)];
            }
            else
            {
                light = keyboard.Key[keyCode];
            }
```
Secondly asus strix scope support, because of the bigger left shift the left window and alt are not in the right place, which are not corrected by the sdk.
```
            // Asus Strix Scope
            if (keyboard.Name == "Charm")
            {
                if (keyCode == 0XDB)
                {//left window
                    light = keyboard.Lights[(int)(5 * keyboard.Width + 2)];
                }
                else if (keyCode == 0x38)
                {//left alt
                    light = keyboard.Lights[(int)(5 * keyboard.Width + 3)];
                }
            }
```

**Comment by Fluto** [comment]
Added a way to test lights for custom peripheral devices. To make life easier
test lights

**Review by diogotr7 — APPROVED** [review]

**Comment by NicTanghe** [comment]
I just get can't find aura SDK and I have to idea where to get it I can only find aura creator on my pc

**Pull request merged**